tools/codegen/api/dispatcher.py

```diff
@@ -61,10 +61,11 @@ def argument(a: Argument) -> DispatcherArgument:
     else:
         assert local.use_c10_dispatcher() is UseC10Dispatcher.with_codegenerated_unboxing_wrapper
         la = legacy_dispatcher.argument(a)
+        assert len(la) == 1, "TensorOptions arguments not supported"
```

Comment:
IIUC, the real invariant here is that when `local.use_c10_dispatcher() is UseC10Dispatcher.with_codegenerated_unboxing_wrapper` then `legacy_dispatcher.argument(a)` is guaranteed to return a single element; it has nothing to do with TensorOptions arguments not being supported.

Although this is ok to leave up to personal preference, I think a better way to handle the problem here is to split `legacy_dispatcher.argument` into two functions: one that is guaranteed to return only a single LegacyDispatcherArgument (but is only valid for `with_codegenerated_unboxing_wrapper`), and then another one which is valid in all cases but might return a sequence.